### DISCUSSION TITLE
How to replace a pixel of a specified color with another color

### DISCUSSION THREAD

**Discussion by wf-soft** [SixLabors/ImageSharp#1502] [Q&A] [answered]
I don't seem to have found an API to replace colors

**Comment by JimBobSquarePants** [comment, marked as answer]
Here's two ways to do it. Available on `Image<TPixel>` and individual frames.

[For a given pixel at position x, y](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Image-1.html#SixLabors_ImageSharp_Image_1_Item_System_Int32_System_Int32_)

```c#
// Image = Image<TPixel>
image[x, y] = ...
```
[For operating on multiple pixels on the same row](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Image-1.html#SixLabors_ImageSharp_Image_1_GetPixelRowSpan_System_Int32_).

```c#
// Image = Image<TPixel>
Span<TPixel> row = image.GetPixelRowSpan(y);
row[x] = ...
```

**Reply by smhoff** [reply]
This would be great, except, where can I find GetPixeRowSpan?

**Reply by JimBobSquarePants** [reply]
That’s V1. See the docs for updated samples. https://docs.sixlabors.com/articles/imagesharp/pixelbuffers.html

**Comment by wf-soft** [comment]
@JimBobSquarePants  I use the following method, but it doesn't seem to work, while mutate works normally.

```
var replace = new Rgba32(250, 250, 250);
            List<Rgba32> colors = new List<Rgba32>();
            colors.AddRange(Enumerable.Range(150, 105).Select(x => new Rgba32(x, x, x)));
            for (var y = 0; y < ImgFile1.ImgEditVM.image.Height; y++)
            {
                var row = ImgFile1.ImgEditVM.image.GetPixelRowSpan(y);
                for (var x = 0; x < ImgFile1.ImgEditVM.image.Width; x++)
                {
                    foreach (var color in colors)
                    {
                        if (row[x] == color) row[x] = replace;
                    }
                }
            }
//ImgFile1.ImgEditVM.image.Mutate(x => x.BinaryThreshold(1));
ImgFile1.ImgEditVM.SaveChang();
```

**Reply by wf-soft** [reply]
Am I missing any steps

**Reply by antonfirsov** [reply]
If I understand your code correctly, I assume there are not so many `(r,g,b) == (x,x,x)` (grayscale) pixels in your image, these are the only colors your code will replace.

Troubleshooting tip:
For making sure you can actually manipulate pixels from your code, I would first try to do something very simple, for example replacing every second row. Then move on implementing your algorithm.

**Reply by wf-soft** [reply]
It seems to work. Thank you for your help
